src/main/java/io/cdap/plugin/gcp/gcs/actions/GCSDoneFileMarker.java

```diff
@@ -0,0 +1,176 @@
+ * pipeline.
+ */
+@Plugin(type = PostAction.PLUGIN_TYPE)
+@Name(GCSDoneFileMarker.NAME)
+@Description("Creates a marker file with a given name in case of a succeeded, failed or completed pipeline.")
+public class GCSDoneFileMarker extends PostAction {
+  private static final Log log = LogFactory.getLog(GCSDoneFileMarker.class);
+  public static final String NAME = "GCSDoneFileMarker";
+  public Config config;
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    config.validate(pipelineConfigurer.getStageConfigurer().getFailureCollector());
+  }
+
+  @Override
+  public void run(BatchActionContext batchActionContext) throws IOException {
+    config.validate(batchActionContext.getFailureCollector());
+    if (!config.shouldRun(batchActionContext)) {
+      log.debug("Action did not get executed. No marker file is created.");
+      return;
+    }
+
+    GCSPath markerFilePath = config.getPath();
+    String serviceAccount = config.getServiceAccount();
+    createFileMarker(config.getProject(), markerFilePath, serviceAccount, config.isServiceAccountFilePath(),
+                     batchActionContext);
+  }
+
+  /**
+   * Config for the plugin.
+   */
+  public static class Config extends GCPConfig {
+    public static final String NAME_PATH = "path";
+    public static final String NAME_RUN_CONDITION = "runCondition";
+
+    @Name(NAME_RUN_CONDITION)
+    @Description("When to run the action. Must be 'completion', 'success', or 'failure'. Defaults to 'completion'. " +
+      "If set to 'completion', the action will be executed and a marker file will get created regardless of whether " +
+      "the pipeline run succeeded or failed. If set to 'success', the action will get executed and the marker file " +
+      "will get created only if the pipeline run succeeded. If set to 'failure', the action will get executed and " +
+      "the marker file will get created only if the pipeline run failed")
+    public String runCondition;
+
+    @Name(NAME_PATH)
+    @Description("GCS path where the marker file will get created.")
+    @Macro
+    private String path;
+
+    Config() {
+      super();
+      this.runCondition = Condition.SUCCESS.name();
+    }
+
+    public GCSPath getPath() {
+      return GCSPath.from(path);
+    }
+
+    void validate(FailureCollector collector) {
+      if (!this.containsMacro(NAME_RUN_CONDITION)) {
+        new ConditionConfig(runCondition).validate(collector);
+      }
+
+      if (!containsMacro(NAME_PATH)) {
+        try {
+          getPath();
+        } catch (IllegalArgumentException e) {
+          collector.addFailure(e.getMessage(), "Please provide a valid GCS path.")
+            .withConfigProperty(NAME_PATH);
+        }
+      }
+
+      Boolean isServiceAccountFilePath = isServiceAccountFilePath();
+      if (isServiceAccountFilePath == null) {
+        collector.addFailure("Service account type is undefined.", "Must be `filePath` or `JSON`")
```

Comment:
`isServiceAccountFilePath` is null, meaning its a macro? Has this case been tested?

Reply:
also if nothing is provided, we should default to filePath.

Reply:
Please make sure following cases are tested:
1.) service account type is macro
2.) service account file path or json value is a macro
3.) service account type not provided. in that case defaults to filepath

Reply:
@CuriousVini All cases are tested now, and the plugin is working correctly. Please check. 

The *missing?* comments are written in an older commit. These are already fixed.